starknet_py/net/signer/base_signer.py

```diff
@@ -20,7 +20,7 @@ def public_key(self) -> int:
         """
 
     @abstractmethod
-    def sign_transaction(self, transaction: Transaction) -> List[int]:
+    def sign_transaction(self, transaction: AccountTransaction) -> List[int]:
```

Comment:
Not sure if this is okay that BaseAccount accepts any `Transaction` in its methods, but BaseSigner only works with `AccountTransaction`.